fix: support distance display in metric units

doncicuto opened the pull request:
Hi there, the following PR deals with issue #116 

I've added a useMetricUnits props to the affected components reported in the issue:

- [X] AccessLegSummary (package itinerary-body)
- [X] AccessLeg (package printable-itinerary)
- [X] TransitStopOption (package location-field)

This prop is optional with default value false which means that imperial units will be used. If useMetricUnits is set, metric units will be used instead.

I've also added some stories to test useMetricUnits in action:

![image](https://github.com/opentripplanner/otp-ui/assets/30386061/5a2470ab-a56d-43d1-8a62-7ac6439b55b9)

As this is my first ever contribution to this project so I hope this first contribution is useful. Please don't hesitate to tell me what I should improve or change for my future contributions. 

This an awesome project, thank's a lot for your effort and the quality of the code. It's great that you've published it as an open source project, hopefully I'll try to upload more contributions soon.

Cheers!

miles-grant-ibigroup reviewed (changes requested):
Thanks so much for contributing! And thanks so much for your kind words about the project. Your prop-based solution is well-implemented, but I think we're hoping to use our localization library to implement this feature

doncicuto commented:
Hi @miles-grant-ibigroup, sorry I didn't know about your localization library. Should I close this PR, do you want me to check that localization library? Let me know if I can be of any help. Thanks

miles-grant-ibigroup commented:
> Hi @miles-grant-ibigroup, sorry I didn't know about your localization library. Should I close this PR, do you want me to check that localization library? Let me know if I can be of any help. Thanks

You can keep this PR open! We use `react-intl`. This exposes an `intl` object that should contain the current locale. Figuring out whether to use metric or imperial based on this locale is the challenge. Perhaps a fixed list of imperial languages (en-US)? 

I hope this is helpful. This is on our backlog to do ourselves, but we probably won't get to it soon.

miles-grant-ibigroup commented:
> Hi there. It seems that only three countries still use the imperial units so I can hardcode those countries and use react-intl. If you are not in a hurry I'd like to work on this issue.

That sounds great! Please do it's much appreciated. I'll be here to review when you're ready :)

doncicuto commented:
Great @miles-grant-ibigroup. How about using this?

```
  // Reference: https://en.wikipedia.org/wiki/Mile
  // While most countries abandoned the mile when switching to the metric system,
  // the international mile continues to be used in some countries, such as Liberia,
  // Myanmar, the United Kingdom and the United States. It is also used in a number
  // of territories with less than a million inhabitants, most of which are UK or US territories,
  // or have close historical ties with the UK or US: American Samoa, Bahamas, Belize,
  // British Virgin Islands, Cayman Islands, Dominica, Falkland Islands, Grenada,Guam,
  // The N. Mariana Islands, Samoa, St. Lucia, St. Vincent & The Grenadines, St. Helena,
  // St. Kitts & Nevis, the Turks & Caicos Islands, and the US Virgin Islands.
  // The mile is even encountered in Canada, though this is predominantly in rail transport
  // and horse racing, as the roadways have been metricated since 1977.
  const useMetricUnits = [
    "en-US",
    "en-GB",
    "en-LR",
    "en-AS",
    "sm-AS",
    "en-BS",
    "en-BZ",
    "es-BZ",
    "en-VG",
    "en-KY",
    "en-DM",
    "en-GD",
    "en-GU",
    "en-MP",
    "en-WS",
    "sm-WS",
    "en-LC",
    "en-VC",
    "en-SH",
    "en-KN",
    "en-TC",
    "en-VI",
    "my-MM"
  ].includes(intl.locale);
```

doncicuto commented:
Hi @miles-grant-ibigroup I've modified the PR adding a function to check if according to react-intl locale we have to use metric or imperial units. Now fewer files are modified.

Here are two screenshots showing the changes of locale in action and how the units are changed.

Default locale (en-US):
![Captura desde 2024-04-26 11-00-58](https://github.com/opentripplanner/otp-ui/assets/30386061/779dd0e1-64d4-4222-a725-d79ebb5bbac6)

Using Spanish:
![Captura desde 2024-04-26 11-01-07](https://github.com/opentripplanner/otp-ui/assets/30386061/74a9353d-a439-4bc4-bcdb-99afd41e2c5b)

Also, hope you don't mind, as I'm a Spaniard I've changed some entries for the itinerary-body es.yaml

Please let me know if you want me to change anything

Cheers!

miles-grant-ibigroup commented on packages/itinerary-body/src/defaults/access-leg-step.tsx:
```diff
@@ -104,10 +105,13 @@ export default function AccessLegStep({
-          {humanizeDistanceString(step.distance, false, intl)}
+          {humanizeDistanceString(
+            step.distance,
+            coreUtils.metricUnits.areMetricUnitsUsed(intl.locale),
+            intl
```
Would it be possible to refactor `humanizeDistanceString` to use the `intl` object it gets anyway to avoid having to import `coreUtils` everywhere?

miles-grant-ibigroup replied:
I'd say don't remove the manual override (for backwards compatibility) but instead use it as an override -- a way to force metric even if the locale isn't metric

miles-grant-ibigroup reviewed (changes requested):
This is a fantastic contribution thank you so much! I really appreciate your translation updates as well. 

I have one review comment that would clean up the code. Once that's done I'll assign to a second reviewer! We do 2 reviews per PR -- once the second reviewer approves we will merge!

miles-grant-ibigroup commented:
As for the failing unit tests running `yarn unit -u` will update those!

doncicuto commented:
Thanks @miles-grant-ibigroup I'll prepare the changes

nukeador commented:
@doncicuto @miles-grant-ibigroup I'm wondering what's the status of this change? I don't want to keep hacking node_module libraries to be able to use metric units on otp and otp-react-redux :-)

miles-grant-ibigroup commented:
> @doncicuto @miles-grant-ibigroup I'm wondering what's the status of this change? I don't want to keep hacking node_module libraries to be able to use metric units on otp and otp-react-redux :-)

There's some changes that we need to be able to merge this PR! If you take a look at the conversation here you might be able to take a swing at it, which I'd greatly appreciate :)

That being said you might want to start a fresh PR, since this one is quite outdated at this point
